view_user_birthdate.py

```diff
@@ -0,0 +1,242 @@
+    Date.
+    """
+    nationalities = ['chinese', 'north korean', 'south korean',
+                     'japanese', 'taiwanese', 'hungarian',
+                     'lithuanian', 'iranian'] # Abbreviated list.
+    def format(self):
+        """Using the attributes 'month', 'day', and 'year', return a
+        string in the Chinese format:
+        yyyy/mm/dd
+        """
+        return '%s/%s/%s' % (self.year, self.month, self.day)
+
+def read_file(file_name):
+    """Read in data from a file and return it."""
+    with open(file_name, 'r') as some_file:
+        data = some_file.read()
+    return data
+
+def chunk_data(file_data, size):
+    """Take in data as a string and an integer as a string. Return a
+    list containing chunks of data that are 'size' number of lines
+    each.
+    """
+    data_lines = file_data.splitlines()
+
+    assert data_lines, "There is no data!"
+    assert int(size), "An integer in the form of a string must be provided."
+    assert 0 < int(size) <= len(data_lines), "A valid size must be provided."
+    assert len(data_lines) % size == 0, "Data is incomplete."
+
+    return [data_lines[pos:pos + size] for pos in range(0, len(data_lines), size)]
+
+def parse(data_chunk):
+    """Take in a chunk of data 3 lines long. Parse the 3 lines of code
+    into their corresponding parts: username, nationality, and
+    birthdate. Then return the parsed parts.
+    """
+    # Parse username.
+    first_line = re.search(r"username:(?P<username>[\w.-]+)", data_chunk[0])
+    if first_line:
+        username = first_line.group('username')
+    else:
+        username = None
+
+    # Parse nationality.
+    second_line = re.search("nationality:"
+                            "(?P<nationality>[a-zA-Z]+ ?[a-zA-Z]*)", data_chunk[1])
+    if second_line:
+        nationality = second_line.group('nationality')
+    else:
+        nationality = None
+
+    # Parse birthdate.
+    last_line = re.search("birthdate:"
+                          "(?P<birthdate>[0-9]{1,2}/[0-9]{1,2}/[0-9]{4})", data_chunk[2])
+    if last_line:
+        birthdate = last_line.group('birthdate')
+    else:
+        birthdate = None
+
+    return username, nationality, birthdate
+
+def build_bday_book(file_name):
+    """Take in the name of a file. Use the contents of the file to
+    build a dictionary where the keys of the dictionary are the
+    usernames and the values are the corresponding birthdates.
+    """
+    data = read_file(file_name)
+    assert data, "There is no data!"
+    data_chunks = chunk_data(data, 3)
+    for chunk in data_chunks:
+        username, nationality, birthdate = parse(chunk)
+        if username and nationality and birthdate:
+            month, day, year = build_date(birthdate)
+            birthdate = format_date(month, day, year, nationality)
+            # Instantiate a User object thus adding the User to the bday_book.
+            User(username, nationality, birthdate)
```

Comment:
constructors shouldn't have side effects - here you're creating a User object then not doing anything with it, entirely relying on the side effect. it'd be better to do something like

bday_book[username] = User(username, nationality, birthdate)

The bday book dictionary can be built outside of user, doesn't need to live inside user.